src/contour/ContourGuiApp.cpp

```diff
@@ -445,6 +443,48 @@ int ContourGuiApp::terminalGuiAction()
     return rv;
 }
 
+void ContourGuiApp::setupQCoreApplication()
+{
+    auto const* profile = _config.profile(profileName());
+    Require(profile);
+
+    auto const defaultAppName = QStringLiteral("contour");
+    auto const defaultOrgDomain = QStringLiteral("contour-terminal.org");
+    auto const defaultOrgName = QStringLiteral("contour");
+
+    auto const platformName = QGuiApplication::platformName();
+
+    if (platformName == "xcb")
+    {
+        // On X11, we want to set the WM_CLASS property to the configured value.
+        QCoreApplication::setOrganizationDomain(defaultOrgDomain);
+        QCoreApplication::setOrganizationName(defaultOrgName);
+        QCoreApplication::setApplicationName(QString::fromStdString(profile->wmClass));
+        QCoreApplication::setApplicationVersion(CONTOUR_VERSION_STRING);
+    }
+    else if (platformName == "wayland")
+    {
+        // On Wayland, we want to set the Applicion Id to the configured value via the desktop file name.
+        // We use the desktop file name as the application id, because that's what Qt uses to set the
+        // app id on Wayland.
+        // I know this sounds weird. This is because it is weird. But it's the only way to set the
+        // application id on Wayland when using Qt.
+        if (profile->wmClass.empty())
+            QGuiApplication::setDesktopFileName(QString::fromStdString(profile->wmClass));
+        QCoreApplication::setOrganizationDomain(defaultOrgDomain);
+        QCoreApplication::setOrganizationName(defaultOrgName);
+        QCoreApplication::setApplicationName(defaultAppName);
+        QCoreApplication::setApplicationVersion(CONTOUR_VERSION_STRING);
+    }
+    else
+    {
+        QCoreApplication::setOrganizationDomain(defaultOrgDomain);
+        QCoreApplication::setOrganizationName(defaultOrgName);
+        QCoreApplication::setApplicationName(defaultAppName);
+        QCoreApplication::setApplicationVersion(CONTOUR_VERSION_STRING);
+    }
+}
+
```

Comment:
We don't need to duplicate 4 same lines for each if statement here

within xcb if check i would set appName to wmClass if it exists

within wayland check i would set desktop file name if wmClass exists

everything else should just default i believe

Reply:
yeah you're right. at first I didn't know how to approach it, so i was writing it a little bit more naively. thanks ;)